vscode-dotnet-runtime-library/src/Utils/WebRequestWorker.ts

```diff
@@ -2,57 +2,164 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
-import retry from 'p-retry';
-import * as request from 'request-promise-native';
+import Axios from 'axios';
+import axiosRetry from 'axios-retry';
+import { AxiosCacheInstance, buildStorage, setupCache, StorageValue } from 'axios-cache-interceptor';
 import { IEventStream } from '../EventStream/EventStream';
 import { WebRequestError, WebRequestSent } from '../EventStream/EventStreamEvents';
 import { IExtensionState } from '../IExtensionState';
+import { Debugging } from '../Utils/Debugging';
+
+/*
+This wraps the VSCode memento state blob into an axios-cache-interceptor-compatible Storage.
+(The momento state is used to save extensionState/data across runs of the extension.)
+All the calls are synchronous.
+*/
+const mementoStorage = (extensionStorage: IExtensionState) => {
+    const cachePrefix = 'axios-cache'; // Used to make it easier to tell what part of the extension state is from the cache
+    return buildStorage({
+        // tslint:disable-next-line
+        set(key: string, value: any) {
+            extensionStorage.update(`${cachePrefix}:${key}`, value);
+        },
+        remove(key: string) {
+            extensionStorage.update(`${cachePrefix}:${key}`, undefined);
+        },
+        find(key: string) {
+            return extensionStorage.get(`${cachePrefix}:${key}`) as StorageValue;
```

Comment:
does extensionstroage return undefined if not found? Thats what the axios cache interceptor expects

Reply:
yes according to vscode docs